src/main/java/org/ibp/api/java/impl/middleware/common/validator/LocationValidator.java

```diff
@@ -65,18 +95,216 @@ public void validateSeedLocationAbbr(final BindingResult errors, final List<Stri
+		this.errors = new MapBindingResult(new HashMap<>(), LocationRequestDto.class.getName());
+
+		this.validateLocationName(locationRequestDto.getName());
+		this.validateLocationType(locationRequestDto.getType());
+		this.validateLocationAbbr(locationRequestDto.getAbbreviation());
+		this.validateLocationAbbrNotExists(locationRequestDto.getAbbreviation());
+
+		if (locationRequestDto.getCountryId() != null) {
+			this.validateLocationId(locationRequestDto.getCountryId(), "location.country.invalid");
+		}
+
+		if (locationRequestDto.getProvinceId() != null) {
+			if (locationRequestDto.getCountryId() == null) {
+				this.errors.reject("location.country.required.when.province.is.defined", "");
+				throw new ApiRequestValidationException(this.errors.getAllErrors());
+			}
+			this.validateLocationId(locationRequestDto.getProvinceId(), "location.province.invalid");
+		}
+
+		if (locationRequestDto.getCountryId() != null && locationRequestDto.getProvinceId() != null && !locationRequestDto.getProvinceId().equals(0)) {
+			this.validateIfProvinceBelongToCountry(locationRequestDto.getCountryId(), locationRequestDto.getProvinceId());
+		}
+	}
+
+	private void validateLocationId(final Integer locationId, final String errorCode) {
+		final LocationDTO locationDTO = this.locationService.getLocation(locationId);
+		if (locationDTO == null) {
+			this.errors.reject(errorCode, "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	private void validateIfProvinceBelongToCountry(final Integer countryId, final Integer provinceId) {
+		final LocationDTO locationDTO = this.locationService.getLocation(provinceId);
+		if (!locationDTO.getCountryId().equals(countryId)) {
+			this.errors.reject("location.province.not.belong.to.country", "");
+			throw new ApiRequestValidationException(this.errors.getAllErrors());
+		}
+	}
+
+	public void validateUpdate(final Integer locationId, final LocationRequestDto locationRequestDto) {
+		this.errors = new MapBindingResult(new HashMap<>(), LocationRequestDto.class.getName());
+
+		this.validateLocation(this.errors, locationId);
+		if (locationRequestDto.getName() != null) {
+			this.validateLocationName(locationRequestDto.getName());
+		}
+
+		final LocationDTO locationDTO = this.locationService.getLocation(locationId);
+		final Integer countryId = locationRequestDto.getCountryId() != null ? locationRequestDto.getCountryId() : locationDTO.getCountryId();
+		final Integer provinceId = locationRequestDto.getProvinceId() != null ? locationRequestDto.getProvinceId() : locationDTO.getProvinceId();
+
+		if (locationRequestDto.getType() != null) {
+			this.validateLocationType(locationRequestDto.getType());
+		}
+
+		if (locationRequestDto.getAbbreviation() != null) {
+			this.validateLocationAbbr(locationRequestDto.getAbbreviation());
+			this.validateLocationAbbrNotExists(locationId, locationRequestDto.getAbbreviation());
+		}
+
+		if (countryId != null) {
```

Comment:
seems like country and province validations are duplicated in update and create, try to create a function and reuse it 
@cuenyad